src/Foundation/DataTransferObject/HasResolvable.php

```diff
@@ -47,15 +48,12 @@ public static function resolveFrom(FormRequest | Model | array $abstract): stati
      * @param  array<TKey, TValue>  $data
      *
      * @throws MappingError
+     *
+     * @deprecated use hydrate(array $data) instead
      */
     public static function resolve(array $data): static
     {
-        /** @var static $instance */
-        $instance = (new MapperBuilder())
-            ->mapper()
-            ->map(signature: static::class, source: static::resolveTheArrayKeyForm(data: $data));
-
-        return $instance;
+        return static::hydrate($data);
     }
```

Comment:
Trait consumption is make compotition schema so wen user use the trait non DTO class no breaking changes, can we put the hydrate back to the trait for now? and we able to clean up it in V3 to remove this trait

Reply:
got it